crates/store/src/db/models/queries.rs

```diff
@@ -0,0 +1,1640 @@
+    borrow::Cow,
+    collections::{BTreeMap, BTreeSet},
+};
+
+use diesel::{
+    JoinOnDsl, NullableExpressionMethods, OptionalExtension, SqliteConnection, alias,
+    prelude::Queryable, query_dsl::methods::SelectDsl,
+};
+use miden_lib::utils::{Deserializable, Serializable};
+use miden_node_proto::domain::account::{AccountInfo, AccountSummary, NetworkAccountPrefix};
+use miden_node_utils::limiter::{
+    QueryParamAccountIdLimit, QueryParamBlockLimit, QueryParamLimiter, QueryParamNoteIdLimit,
+    QueryParamNoteTagLimit, QueryParamNullifierLimit, QueryParamNullifierPrefixLimit,
+};
+use miden_objects::{
+    Felt, Word,
+    account::{
+        Account, AccountDelta, AccountId, AccountStorageDelta, AccountVaultDelta,
+        FungibleAssetDelta, NonFungibleAssetDelta, NonFungibleDeltaAction, StorageMapDelta,
+        StorageSlot,
+        delta::{AccountUpdateDetails, LexicographicWord},
+    },
+    asset::{Asset, NonFungibleAsset},
+    block::{BlockAccountUpdate, BlockHeader, BlockNoteIndex, BlockNumber},
+    crypto::{hash::rpo::RpoDigest, merkle::MerklePath},
+    note::{NoteExecutionMode, NoteId, NoteInclusionProof, Nullifier},
+    transaction::OrderedTransactionHeaders,
+};
+
+use super::{
+    super::models, BoolExpressionMethods, DatabaseError, NoteSyncRecordRawRow, QueryDsl,
+    RunQueryDsl, SelectableHelper,
+};
+use crate::{
+    db::{
+        NoteRecord, NoteSyncRecord, NoteSyncUpdate, NullifierInfo, Page, StateSyncUpdate,
+        TransactionSummary,
+        models::{
+            AccountRaw, AccountSummaryRaw, BigIntSum, ExpressionMethods, NoteRecordRaw,
+            TransactionSummaryRaw, block_number_to_raw_sql,
+            conv::{
+                aux_to_raw_sql, consumed_to_raw_sql, execution_hint_to_raw_sql,
+                execution_mode_to_raw_sql, fungible_delta_to_raw_sql, idx_to_raw_sql,
+                network_account_prefix_to_raw_sql, nonce_to_raw_sql, note_type_to_raw_sql,
+                nullifier_prefix_to_raw_sql, raw_sql_to_idx, raw_sql_to_nonce, raw_sql_to_slot,
+                slot_to_raw_sql, tag_to_raw_sql,
+            },
+            deserialize_raw_vec, get_nullifier_prefix, raw_sql_to_block_number, serialize_vec,
+            sql_sum_into, vec_raw_try_into,
+        },
+        schema,
+    },
+    errors::{NoteSyncError, StateSyncError},
+};
+
+/// Select notes matching the tags and account IDs search criteria using the given [Connection].
+///
+/// # Returns
+///
+/// All matching notes from the first block greater than `block_num` containing a matching note.
+/// A note is considered a match if it has any of the given tags, or if its sender is one of the
+/// given account IDs. If no matching notes are found at all, then an empty vector is returned.
+///
+/// # Note
+///
+/// This method returns notes from a single block. To fetch all notes up to the chain tip,
+/// multiple requests are necessary.
+pub(crate) fn select_notes_since_block_by_tag_and_sender(
+    conn: &mut SqliteConnection,
```

Comment:
Am I understanding correctly that this represents either a database connection, or a database transaction? And its up to the caller to ensure its the right one?